planner/core/preprocess.go

```diff
@@ -489,22 +488,16 @@ func checkColumn(colDef *ast.ColumnDef) error {
 			return types.ErrTooBigFieldLength.GenWithStack("Column length too big for column '%s' (max = %d); use BLOB or TEXT instead", colDef.Name.Name.O, mysql.MaxFieldCharLength)
 		}
 	case mysql.TypeVarchar:
-		maxFlen := mysql.MaxFieldVarCharLength
-		cs := tp.Charset
-		// TODO: TableDefaultCharset-->DatabaseDefaultCharset-->SystemDefaultCharset.
-		// TODO: Change TableOption parser to parse collate.
-		// Reference https://github.com/pingcap/tidb/blob/b091e828cfa1d506b014345fb8337e424a4ab905/ddl/ddl_api.go#L185-L204
+		// Here logical move backend processing, need to get all levels of character set to set the character set of column.
+		// It's not easy to get the schema charset and table charset here.
+		// ColumnDefaultCharset --> TableDefaultCharset-->DatabaseDefaultCharset-->SystemDefaultCharset.
 		if len(tp.Charset) == 0 {
-			cs = mysql.DefaultCharset
+			return nil
```

Comment:
// It's not easy to get the schema charset and table charset here. The charset is determined by the order `ColumnDefaultCharset --> TableDefaultCharset-->DatabaseDefaultCharset-->SystemDefaultCharset`.
// return nil, to make the check in the ddl.CreateTable.

Reply:
Done.